spec/spec_helper.rb

```diff
@@ -33,6 +33,102 @@ def compatible_request(method, action, hash = {})
   end
 end
 
+
+#
+# Matcher to validate the hierarchy of fields described in an internal 'returns' object (without checking their type)
+#
+# For example, code such as:
+#           returns_obj = Apipie.get_resource_description(...)._methods.returns.detect{|e| e.code=200})
+#           expect(returns_obj).to match_param_structure([:pet_name, :animal_type, :pet_measurements => [:weight, :height]])
+#
+# will verify that the payload structure described for the response of return code 200 is:
+#           {
+#             "pet_name": <any>,
+#             "animal_type": <any>,
+#             "pet_measurements": {
+#                 "weight": <any>,
+#                 "height": <any>
+#             }
+#           }
+#
+#
+RSpec::Matchers.define :match_field_structure do |expected|
```

Comment:
Were are we using those?

Reply:
Used quite heavily in spec/lib/swagger/swagger_dsl_spec.rb